mlflow/utils/autologging_utils/__init__.py

```diff
@@ -376,11 +376,13 @@ def autolog(*args, **kwargs):
             except Exception:
                 pass
 
+            # revert existing patches, will install new patch in following
+            revert_patches(name)
```

Comment:
I move the `revert_patches(name)` to front. Then it will revert all existing patches before setting up new patches. This is due to the strict check I add it [here](https://github.com/mlflow/mlflow/pull/4628/files#r679979273).